.github/workflows/firebase-hosting-merge.yml

```diff
@@ -11,12 +11,25 @@ jobs:
     name: Build and Deploy
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Install Dependencies
-        run: yarn install
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18
+
+      - name: Install pnpm
+        run: corepack enable && corepack prepare pnpm@latest --activate
+
+      - name: Install dependencies
+        run: pnpm install
+      
       - name: Build the application
-        run: yarn run build
-      - uses: FirebaseExtended/action-hosting-deploy@v0
+        run: pnpm build
+
+      - name: Deploy on Firebase
```

Comment:
Try integrating `process.env` for firebase API keys

Reply:
Complexity high since the `nx` workspace does not directly support the `process` variable in angular. It needs a [Custom Webpack Config](https://nx.dev/recipes/environment-variables/use-environment-variables-in-angular) to do so